### PR TITLE
Warn for lambda defaults/`params` differences

### PR DESCRIPTION
**Pull request by jjonescz**
Changes error to warning when lambda parameter default values or `params` modifiers differ during conversion.

Relates to https://github.com/dotnet/roslyn/pull/64961

### CODE REVIEW

File: proposals/lambda-method-group-defaults.md [hunk 1]
```diff
@@ -216,6 +216,13 @@ Same rules as for method parameters ([§14.6.2](https://github.com/dotnet/csharp
 
 No changes to the grammar are necessary for method groups since this proposal would only change their semantics.
 
+The following addition (in bold) is required to anonymous function conversions ([§10.7](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/conversions.md#107-anonymous-function-conversions)):
+
+> Specifically, an anonymous function `F` is compatible with a delegate type `D` provided:
+>
+> - [...]
+> - If `F` has an explicitly typed parameter list, each parameter in `D` has the same type and modifiers as the corresponding parameter in `F` **(`params` modifiers and default values are not considered)**.
```
Comment (cston) [inline proposals/lambda-method-group-defaults.md:224]: >**(`params` modifiers and default values are not considered)** [](http://example.com/codeflow?start=143&length=62)

Or alternatively perhaps: `**ignoring `params` modifiers and default values**`

**Review by cston — APPROVED** [review]

**Comment by jcouv** [comment]
FYI, I'm hoping to get a firm/recorded decision from LDM today.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by jcouv** [comment]
Yay. We got this all sorted out. This can be merged. Thanks for your patience.

**Pull request merged**